stats/stats_test.go

```diff
@@ -281,7 +283,10 @@ func (te *test) doUnaryCall(c *rpcConfig) (*testpb.SimpleRequest, *testpb.Simple
 	} else {
 		req = &testpb.SimpleRequest{Id: errorID}
 	}
-	ctx := metadata.NewOutgoingContext(context.Background(), testMetadata)
+
+	tCtx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	ctx := metadata.NewOutgoingContext(tCtx, testMetadata)
```

Comment:
This call to `metadata.NewOutgoingContext()` could be inlined within the `UnaryCall`, just like how it is done for the `FullDuplexCall`.